### DISCUSSION TITLE
Make forge--buffer-draft-p a configuration option

### DISCUSSION THREAD

**Discussion by DamienCassou** [magit/forge#509] [Ideas]
The variable `forge--buffer-draft-p` controls whether the PR to be created will be a draft or not. In some projects, all new PRs should be drafts at first. As a result, it would be helpful to set `forge--buffer-draft-p` as a directory-local variable in these projects. This is already possible but because the variable is private (indicated by the `--` in the name) and because it is not a `defcustom`, this feels awkward.

I suggest creating a `defcustom` for this variable and making it public.

**Comment by DamienCassou** [comment]
I volunteer to create a PR.

**Reply by tarsius** [reply]
Please go ahead with your plan. Thanks!

**Reply by DamienCassou** [reply]
Done in #517.

**Comment by isadon** [comment]
How do I enable this in my config. I am not looking necessary to set this as a directory local variable. I put `(setq forge-buffer-draft-p t)` in my config (had to do it before loading magit) but when creating the pull request (`C-c C-e` shows `t` for `d Create draft`) it gets created as a Normal one.

**Reply by DamienCassou** [reply]
> I put (setq forge-buffer-draft-p t) in my config

`forge-buffer-draft-p` is a buffer local variable as you can see in [forge-post.el](https://github.com/magit/forge/blob/main/lisp/forge-post.el#L139):

```emacs
(make-variable-buffer-local 'forge-buffer-draft-p)
```

This means that each buffer as its own value for this variable and `(setq forge-buffer-draft-p t)` only affects the current one. If you want to give a default value for this variable, you should use `setq-default`:

```emacs
(setq-default forge-buffer-draft-p t)
```

> `C-c C-e` shows `t` for "d Create draft"

This is very surprising to me and I have no explanation. If my understanding of buffer local variables is correct, you should have seen `nil` here instead of `t` unless you have more configurations that the `setq`. But then, if you see `t`, you should definitely get your PR in draft mode if the package supports the forge you are using.

**Reply by tarsius** [reply]
> This is very surprising to me and I have no explanation.

I don't think this ever worked. We only use the buffer-local value, but setting the global option does not cause the local value to be set, only using the toggle command did that.

I have removed the option in d33aaf4d5037d35a7daebd4754314cd22fc9680b.  The commit message explains how to use `forge-edit-post-hook` for this.
